src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -241,6 +243,25 @@ int sd_dhcp_client_id_to_string(const void *data, size_t len, char **ret) {
         return 0;
 }
 
+int sd_dhcp_client_set_state_callback(
+                sd_dhcp_client *client,
+                sd_dhcp_client_callback_t cb,
+                void *userdata) {
+
+        assert_return(client, -EINVAL);
+
+        client->state_callback = cb;
+        client->state_userdata = userdata;
+
+        return 0;
+}
+
+int sd_dhcp_client_get_state(sd_dhcp_client *client) {
```

Comment:
Also this one, and dhcp6_client variants.

Reply:
Done. Same comment as above.